tools/packaging/helm-chart/kata-deploy/templates/kata-gpu-feature-discovery.yaml

```diff
@@ -0,0 +1,77 @@
+{{- if .Values.kataGpuFeatureDiscovery -}}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kata-gpu-feature-discovery
+  namespace: {{ .Release.Namespace }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kata-gpu-feature-discovery
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: kata-gpu-feature-discovery
+    spec:
+      runtimeClassName: kata-qemu-nvidia-gpu
+      nodeSelector:
+        katacontainers.io/gpu-feature-discovery: required
+{{- with .Values.kataDeploy.imagePullSecrets }}
+      imagePullSecrets:
+{{- toYaml . | nindent 6 }}
+{{- end }}
+      initContainers:
+        - image: nvcr.io/nvidia/k8s-device-plugin:v0.17.0
+          name: gpu-feature-discovery
+          command: ["/usr/bin/gpu-feature-discovery"]
+          resources:
+            requests:
+              "nvidia.com/pgpu": 1
+            limits:
+              "nvidia.com/pgpu": 1
+          volumeMounts:
+            - name: output-dir
+              mountPath: "/etc/kubernetes/node-feature-discovery/features.d"
+            - name: host-sys
+              mountPath: "/sys"
+          securityContext:
+            privileged: true
+          env:
+            - name: MIG_STRATEGY
+              value: none
+            - name: GFD_ONESHOT
+              value: "true"
+      containers:
+        - command:
+          - /bin/bash
+          - -c
+          - |
+            GPU_COUNT=$(kubectl get nodes "$NODE_NAME" -o=jsonpath='{.status.allocatable.nvidia\.com/pgpu}')
+            sed -i "s/gpu\.count=1/gpu\.count=$GPU_COUNT/g" /etc/kubernetes/node-feature-discovery/features.d/gfd
+            kubectl label node --overwrite "$NODE_NAME" katacontainers.io/gpu-feature-discovery=completed
+          volumeMounts:
+            - name: output-dir
+              mountPath: "/etc/kubernetes/node-feature-discovery/features.d"
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: spec.nodeName
+          image: '{{ .Values.kataDeploy.repository }}:{{ .Values.kataDeploy.version }}'
+          imagePullPolicy: '{{ .Values.kataDeploy.imagePullPolicy }}'
+          name: post-process
+          securityContext:
+            privileged: true
```

Comment:
This is only needed because we're labeling the nodes, having a privileged container with kubectl means cluster-admin. I do not know if kata-deploy is the right place to run this.

Reply:
What is the longer term path ?

Reply:
> This is only needed because we're labeling the nodes, having a privileged container with kubectl means cluster-admin. I do not know if kata-deploy is the right place to run this.

Note that we already do this kata-deploy. It already has the SA and kubectl.

Reply:
> What is the longer term path ?
The option suggested by @cdesiniotis 

_Update kata-manager to spin up a GFD pod and run it as one-shot. The kata-manager will first detect how many GPUs are on the node, by running lspci, and update the GFD pod spec to request all GPUs.
Update the kata-manager Role to allow for it to create pods in the GPU Operator namespace. https://github.com/NVIDIA/gpu-operator/blob/main/assets/state-kata-manager/0200_role.yaml_